docs/architecture/0002-routing-mk2.md

```diff
@@ -0,0 +1,109 @@
+# 2. Routing Mk2
+
+Date: 2018-12-20
+
+## Status
+
+Accepted
+
+## Context
+
+Routing is a system to allow a user to describe conditions as to where the survey respondent should go after they complete the answers on a question page.
+
+As these conditions can be any boolean expression we need a more complex routing structure to handle 
+
+## Decision
+
+Implement a new version of the routing in the API that can handle nested expressions.
+
+This is heavily inspied by https://github.com/ONSdigital/eq-author-app/wiki/Routing,-MK2
+
+### Terminology
+- Routing - all rules etc for a page with a fallback destination when no rules match.
+- Rule - a series of expressions that when evaluated to true will go to the destination specified.
```

Comment:
More of a question than anything else but do all rules have at least one `ExpressionGroup`? The example structure below has an ANDed group even though it's really a single expression?

Reply:
Yeah a rule has to have at least one group, which is just set to AND by default it just won't have any effect. Saves having to do different things depending on whether there's one expression or two or more expressions.